src/pyj/read_book/settings.pyj

```diff
@@ -4,23 +4,24 @@ from __python__ import hash_literals
 
 from elementmaker import E
 from read_book.globals import runtime
+from session import defaults
 
 opts = {}
 
 def update_settings(settings):
-    settings = settings or {}
-    opts.columns_per_screen = settings.columns_per_screen or {'portrait':0, 'landscape':0}
-    opts.margin_left = max(0, settings.margin_left or 0)
-    opts.margin_right = max(0, settings.margin_right or 0)
+    settings = settings or defaults
+    opts.base_font_size = max(8, min(settings.base_font_size, 64))
+    opts.bg_image_fade = settings.bg_image_fade or 'transparent'
     opts.color_scheme = settings.color_scheme
-    opts.base_font_size = max(8, min(settings.base_font_size or 16, 64))
-    opts.user_stylesheet = settings.user_stylesheet or ''
-    opts.hide_tooltips = settings.hide_tooltips
+    opts.columns_per_screen = settings.columns_per_screen
     opts.cover_preserve_aspect_ratio = v'!!settings.cover_preserve_aspect_ratio'
-    opts.bg_image_fade = settings.bg_image_fade or 'transparent'
-    opts.paged_wheel_scrolls_by_screen = v'!!settings.paged_wheel_scrolls_by_screen'
+    opts.hide_tooltips = settings.hide_tooltips
     opts.is_dark_theme = v'!!settings.is_dark_theme'
-    opts.override_book_colors = settings.override_book_colors or 'never'
+    opts.margin_left = max(0, settings.margin_left)
+    opts.margin_right = max(0, settings.margin_right)
+    opts.override_book_colors = settings.override_book_colors
+    opts.paged_wheel_scrolls_by_screen = v'!!settings.paged_wheel_scrolls_by_screen'
+    opts.user_stylesheet = settings.user_stylesheet
 
```

Comment:
the original code defends against invalid values being set on the passed in object, not just against no settings. I'd rather not lose that.

Reply:
I may be misunderstanding, but the previous code only guarded against values missing from settings with `or`, not invalid values. All the value checks (min/max, etc.) are still present.

I can improve it a bit though by merging the passed in settings with the defaults to ensure there are no missing properties. It would then handle the missing value case as before.